bsp/src/org/jetbrains/bsp/project/resolver/BspProjectResolver.scala

```diff
@@ -271,6 +270,15 @@ object BspProjectResolver {
       .catchBspErrors
       .reportFinished(reporter, sourcesEventId, message, BspBundle.message("bsp.resolver.request.failed.buildtarget.sources"))
 
+    val jvmEnvironment = {
```

Comment:
jvm environment should be primarily target-scoped in BSP. IDEA's "Project JDK" can be considered just a conveience setting. We could have an optional endpoint for a workspace-wise environment, but I'm not sure it's necessary. We could just infer a default Project JDK from individual modules, or use whatever is available.